cirq/pasqal/pasqal_device.py

```diff
@@ -77,21 +78,31 @@ def decompose_operation(self,
             raise TypeError("{!r} is not a gate operation.".format(operation))
 
         # Try to decompose the operation into elementary device operations
-        if not PasqalDevice.is_pasqal_device_op(operation):
-            decomposition = cirq.protocols.decompose(
-                operation, keep=PasqalDevice.is_pasqal_device_op)
+        if not self.is_pasqal_device_op(operation):
+            decomposition = PasqalConverter().pasqal_convert(
+                operation, keep=self.is_pasqal_device_op)
 
         return decomposition
 
-    @staticmethod
-    def is_pasqal_device_op(op: cirq.ops.Operation) -> bool:
+    def is_pasqal_device_op(self, op: cirq.ops.Operation) -> bool:
 
         if not isinstance(op, cirq.ops.Operation):
             raise ValueError('Got unknown operation:', op)
-        return (len(op.qubits) > 1) or isinstance(
-            op.gate, (cirq.ops.IdentityGate, cirq.ops.MeasurementGate,
-                      cirq.ops.PhasedXPowGate, cirq.ops.XPowGate,
-                      cirq.ops.YPowGate, cirq.ops.ZPowGate))
+
+        valid_op = isinstance(op.gate,
+                              (cirq.ops.IdentityGate, cirq.ops.MeasurementGate,
+                               cirq.ops.PhasedXPowGate, cirq.ops.XPowGate,
+                               cirq.ops.YPowGate, cirq.ops.ZPowGate))
+
+        if not valid_op:  # To prevent further checking if already passed
+            if isinstance(
+                    op.gate,
+                (cirq.ops.HPowGate, cirq.ops.CNotPowGate, cirq.ops.CZPowGate,
+                 cirq.ops.CCZPowGate, cirq.ops.CCXPowGate)):
+                expo = op.gate.exponent
+                valid_op = np.isclose(expo, np.around(expo, decimals=0))
```

Comment:
How would this work with parametrized gates? The type of exponent is `Union[float, sympy.Basic]` - it could be parametrized. I think you should make sure you're handling that here and cover that flow with tests.

Reply:
You're right, I made the appropriate changes.